mithril-common/src/test_utils/certificate_chain_builder.rs

```diff
@@ -760,32 +789,66 @@ mod test {
 
     #[test]
     fn builds_certificate_chain_correctly_chained() {
-        let certificates = vec![
-            Certificate {
-                epoch: Epoch(1),
-                ..fake_data::certificate("cert-1".to_string())
-            },
-            Certificate {
-                epoch: Epoch(2),
-                ..fake_data::certificate("cert-2".to_string())
-            },
+        fn create_fake_certificate(epoch: Epoch, index: u64) -> Certificate {
             Certificate {
-                epoch: Epoch(3),
-                ..fake_data::certificate("cert-3".to_string())
-            },
+                epoch,
+                signed_message: format!("certificate-{index}"),
+                ..fake_data::certificate("cert-fake".to_string())
+            }
+        }
+
+        let certificate_1 = create_fake_certificate(Epoch(1), 1);
+        let certificate_2 = create_fake_certificate(Epoch(2), 2);
+        let certificate_3 = create_fake_certificate(Epoch(2), 3);
+        let certificate_4 = create_fake_certificate(Epoch(3), 4);
+        let certificate_5 = create_fake_certificate(Epoch(4), 5);
+        let certificate_6 = create_fake_certificate(Epoch(4), 6);
+        let certificate_7 = create_fake_certificate(Epoch(4), 7);
+        let certificates = vec![
+            certificate_1,
+            certificate_2,
+            certificate_3,
+            certificate_4,
+            certificate_5,
+            certificate_6,
+            certificate_7,
         ];
```

Comment:
Maybe we could use a more compact code since those variables are not used afterward, I don't think that we loose any readability since the index is still provided (maybe giving the index as the first parameter would be a little more clear).
```suggestion
        let certificates = vec![
            create_fake_certificate(Epoch(1), 1),
            create_fake_certificate(Epoch(2), 2),
            create_fake_certificate(Epoch(2), 3),
            create_fake_certificate(Epoch(3), 4),
            create_fake_certificate(Epoch(4), 5),
            create_fake_certificate(Epoch(4), 6),
            create_fake_certificate(Epoch(4), 7),
        ];
```

Reply:
I compacted the code as you propose :+1: